scripts/o3de/o3de/android_support.py

```diff
@@ -0,0 +1,2116 @@
+        :param key_alias:       The alias of the private signing key in the key store
+        :param key_password:    The password to the private signing key in the key store referenced by the key alias
+        """
+
+        self._store_file = Path(store_file)
+        self._store_password = store_password
+        self._key_alias = key_alias
+        self._key_password = key_password
+
+    def to_template_string(self, tabs) -> str:
+        """
+        Generate a string that represents a signing config section that can be inserted into a gradle script.
+
+        :param tabs:    The number of leading tabs for each line to insert for better readability of the gradle script
+        :return:    The signing config string section to insert
+        """
+        tab_prefix = ' '* 4 * tabs  # 4 spaces per tab
+        return f"{tab_prefix}storeFile file('{self._store_file.as_posix()}')\n" \
+               f"{tab_prefix}storePassword '{self._store_password}'\n" \
+               f"{tab_prefix}keyPassword '{self._key_password}'\n" \
+               f"{tab_prefix}keyAlias '{self._key_alias}'\n"
+
+
+JAVA_VERSION_REGEX = re.compile(r'.*(\w)\s(version)\s*\"?([\d\_\.]+)', re.MULTILINE)
+
+def validate_java_environment() -> str:
+    """
+    Java is required in order to build android projects with gradle. Check if java is either on the system PATH
+    environment, or set by the JAVA_HOME environment variable and return the version string. This is the same
+    java search approach used by the sdkmanager.
+    """
+
+    java_home = os.getenv('JAVA_HOME', None)
+    if java_home:
+        java_exe_working_dir = os.path.join(java_home,'bin')
+        java_exe_path = os.path.join(java_exe_working_dir, f'java{EXE_EXTENSION}')
+        if not os.path.isfile(java_exe_path):
+            raise AndroidToolError(f"JAVA_HOME is set to an invalid location ({java_home}). There is no java executable located in that path.")
+    else:
+        java_exe_working_dir = None
+
+    result = subprocess.run([f'java{EXE_EXTENSION}', '-version'],
+                            shell=(platform.system() == 'Windows'),
+                            capture_output=True,
+                            encoding=DEFAULT_READ_ENCODING,
+                            errors=ENCODING_ERROR_HANDLINGS,
+                            cwd=java_exe_working_dir)
+
+    if result.returncode != 0:
+        if java_exe_working_dir:
+            raise AndroidToolError(f"Unable to determine java version from {java_exe_working_dir} ({result.stderr or result.stdout})")
+        else:
+            raise AndroidToolError(f"Unable to locate java. Either set it in the PATH environment or set JAVA_HOME to a valid java installation. ({result.stderr or result.stdout})")
+
+    java_version_match = JAVA_VERSION_REGEX.search(result.stdout or result.stderr)
+    if java_version_match is None:
+        raise AndroidToolError(f"Unable to determine java version")
+
+    java_version = java_version_match.group(3)
```

Comment:
If a named group is used for the java_version_match regex, then it could be extracted using a group name of "version".
This does allow the code to be slightly more maintainable in case the regex above needs to be adjusted in the future. Such as if the Java version format changes in a newer version, then this code can still just extract the "version" part of the regex.
```suggestion
    java_version = java_version_match.group('version')
```